src/main/java/net/oauth/jsontoken/AbstractJsonTokenParser.java

```diff
@@ -0,0 +1,222 @@
+        .getAsJsonObject();
+
+    JsonToken jsonToken = new JsonToken(header, payload, clock, tokenString);
+    return jsonToken;
+  }
+
+  /**
+   * Verifies that the jsonToken has a valid signature and valid standard claims
+   * (iat, exp). Does not need VerifierProviders because verifiers are passed in
+   * directly.
+   *
+   * @param jsonToken the token to verify
+   * @throws SignatureException when the signature is invalid
+   * @throws IllegalStateException when exp or iat are invalid
+   */
+  public void verify(JsonToken jsonToken, List<Verifier> verifiers) throws SignatureException {
+    if (! signatureIsValid(jsonToken.getTokenString(), verifiers)) {
+      throw new SignatureException("Invalid signature for token: " +
+          jsonToken.getTokenString());
+    }
+
+    Instant issuedAt = jsonToken.getIssuedAt();
+    Instant expiration = jsonToken.getExpiration();
+
+    if (issuedAt == null && expiration != null) {
+      issuedAt = new Instant(0);
+    }
+
+    if (issuedAt != null && expiration == null) {
+      expiration = new Instant(Long.MAX_VALUE);
+    }
+
+    if (issuedAt != null && expiration != null) {
+      if (issuedAt.isAfter(expiration)
+          || ! clock.isCurrentTimeInInterval(issuedAt, expiration)) {
+        throw new IllegalStateException(String.format("Invalid iat and/or exp. iat: %s exp: %s "
+            + "now: %s", jsonToken.getIssuedAt(), jsonToken.getExpiration(), clock.now()));
+      }
+    }
+
+    if (checkers != null) {
+      for (Checker checker : checkers) {
+        checker.check(jsonToken.getPayloadAsJsonObject());
+      }
+    }
+  }
+
+  /**
+   * Verifies that a JSON Web Token's signature is valid.
+   *
+   * @param tokenString the encoded and signed JSON Web Token to verify.
+   * @param verifiers used to verify the signature. These usually encapsulate
+   *        secret keys.
+   */
+  public boolean signatureIsValid(String tokenString, List<Verifier> verifiers) {
+    String[] pieces = splitTokenString(tokenString);
+    byte[] signature = Base64.decodeBase64(pieces[2]);
+    String baseString = JsonTokenUtil.toDotFormat(pieces[0], pieces[1]);
+
+    boolean sigVerified = false;
+    for (Verifier verifier : verifiers) {
+      AsciiStringVerifier asciiVerifier = new AsciiStringVerifier(verifier);
+      try {
+        asciiVerifier.verifySignature(baseString, signature);
+        sigVerified = true;
+        break;
+      } catch (SignatureException e) {
+        continue;
+      }
+    }
+    return sigVerified;
+  }
+
+  /**
+   * Verifies that a JSON Web Token is not expired.
+   *
+   * @param jsonToken the token to verify
+   * @param now the instant to use as point of reference for current time
+   * @return false if the token is expired, true otherwise
+   */
+  public boolean expirationIsValid(JsonToken jsonToken, Instant now) {
+    Instant expiration = jsonToken.getExpiration();
+    if ((expiration != null) && now.isAfter(expiration)) {
```

Comment:
Replace this with:

return expiration == null || expiration.isAfter(now);

Reply:
Changed